Update README and rewrite contributing guidelines

tianyizheng02 opened the pull request:
Update Python version badge in README.md from >= 3.6 to 3.12

Update usage examples in README.md to reflect updated API functions

Rewrite CONTRIBUTING.md to extensively document how to set up the development environment and install required/recommended tooling

(I'm not sure what to do about the build status badge in the README, since it doesn't seem like we're using Travis at the moment.)

tianyizheng02 committed:
Update README and rewrite contributing guidelines

Update Python version badge in README.md from >= 3.6 to 3.12

Update usage examples in README.md to reflect updated API functions

Rewrite CONTRIBUTING.md to extensively document how to set up the
development and install required/recommended tooling

RitwikGupta commented on CONTRIBUTING.md:
```diff
@@ -1,4 +1,85 @@
+
+### Code Styling
+
+We don't yet require our contributors to run any code formatters as part of our pre-commit workflow or our GitHub workflows.
```
Just require black. Keep all the code consistent.

tianyizheng02 replied:
Fine by me, l can add it to our GitHub workflows and pre-commit hooks in another PR.

RitwikGupta replied:
Make another PR and link it to this one

tianyizheng02 replied:
By the way, it looks like we're already using flake8 for linting in our GitHub Actions. How about we add that as a requirement as well?

RitwikGupta replied:
sgtm

tianyizheng02 replied:
Added black and flake8 in #174, though flake8 is extremely limited for the time being based on our current configs

tianyizheng02 replied:
@RitwikGupta Added black and flake8 to the repo in #174

tianyizheng02 committed:
Require black in contributing guidelines and fix typos

Update CONTRIBUTING.md to require contributors to use black rather than
simply encourage them to use it (see comments on PR #170). black will be
added as a pre-commit hook and GitHub workflow in a future PR.

RitwikGupta commented on README.md:
```diff
@@ -2,78 +2,71 @@
 ![Build Status](https://travis-ci.org/Pitt-CSC/PittAPI.svg?branch=master)
 ![License](https://img.shields.io/badge/license-GPLv2-blue.svg)
-![Python Version](https://img.shields.io/badge/python-%3E%3D%203.6-green.svg)
+![Python Version](https://img.shields.io/badge/python-3.12-green.svg)
```
Does this really require Python >= 3.12? It seems limiting to support only the latest version of Python with the library. Please lower this accordingly to the actual lowest supported version of Python.

tianyizheng02 replied:
I assumed that the repo now only supports 3.12, since that's what the rest of the repo explicitly calls for (e.g., the README, the Pipfile, and the GitHub Actions workflows). #144 is the PR in which the project was officially migrated to 3.12, and I haven't tested the code with any older versions. @nij-patel, do you know if PittAPI now _only_ supports 3.12, or is it still compatible with some older versions?

tianyizheng02 replied:
Instead of keeping this PR blocked, how about I open a new issue for updating the supported version and leave this as work for a future PR?

tianyizheng02 replied:
Determined actual lowest supported Python version (3.9) in #178

tianyizheng02 replied:
Updated the Python version badge to read ">= 3.9"

tianyizheng02 committed:
Add flake8 to contributing guidelines

tianyizheng02 committed:
Fix typo in filepath

tianyizheng02 commented on README.md:
```diff
@@ -2,78 +2,71 @@
 
 ![Build Status](https://travis-ci.org/Pitt-CSC/PittAPI.svg?branch=master)
```
Also, it doesn't look like we use Travis anymore, so is it OK to remove this badge?
```suggestion
```

nij-patel replied:
yeah, I think so? might need Alex or Ritwik to comment just to be sure

RitwikGupta replied:
Go for it, though someone should probably set up Travis again

tianyizheng02 replied:
Replaced the Travis badge with a GitHub Actions badge

tianyizheng02 committed:
Update Python version badge in README

tianyizheng02 committed:
Replace Travis CI badge in README with GitHub Actions badge

Replace Travis CI badge with GitHub Actions badge since we no longer use
Travis and are now using GitHub actions

Delete .travis.yml config file for Travis since we no longer use Travis
